src/map/status.c

```diff
@@ -12278,38 +12279,118 @@ bool status_readdb_sizefix(char* fields[], int columns, int current)
 	return true;
 }
 
-bool status_readdb_refine(char* fields[], int columns, int current)
-{
-	int i, bonus_per_level, random_bonus, random_bonus_start_level;
-
-	current = atoi(fields[0]);
-
-	if (current < 0 || current >= REFINE_TYPE_MAX)
-		return false;
-
-	bonus_per_level = atoi(fields[1]);
-	random_bonus_start_level = atoi(fields[2]);
-	random_bonus = atoi(fields[3]);
-
-	for(i = 0; i < MAX_REFINE; i++)
-	{
-		char* delim;
-
-		if (!(delim = strchr(fields[4+i], ':')))
-			return false;
-
-		*delim = '\0';
-
-		status->dbs->refine_info[current].chance[i] = atoi(fields[4+i]);
+/**
+ * Processes a refine_db.conf entry.
+ *
+ * @param *r	  Libconfig setting entry. It is expected to be valid and it
+ *                won't be freed (it is care of the caller to do so if
+ *                necessary)
+ * @param n       Ordinal number of the entry, to be displayed in case of
+ *                validation errors.
+ * @param *source Source of the entry (file name), to be displayed in case of
+ *                validation errors.
+ * @return # of the validated entry, or 0 in case of failure.
+ */
+int status_readdb_refine_libconfig_sub(config_setting_t *r, const char *name, const char *source) {
+	config_setting_t *rate = NULL;
+	int i=0, type=0, bonus_per_level=0, rnd_bonus_v=0, rnd_bonus_lv=0;
+	
+	if(!strncmp(name, "Armors", 6))
+		type = 0;
+	else if(!strspn(&name[strlen(name)-1], "0123456789") || !(type = atoi(strtok(strdup(name), "WeaponLevel")))) {
```

Comment:
You're calling `strdup`, and never freeing its result. While this doesn't show up in the builtin memory leak detector (you're working around it by calling `strdup` directly instead of `aStrdup`), it still leaks memory.
You can avoid this entirely, by checking the string length: if the string is longer than the length of `"WeaponLevel"`, then you can `strncmp` it against `"WeaponLevel"` (using the appropriate max length), and then call `atoi` on the fist index after that.